main.tf

```diff
@@ -33,3 +33,52 @@ resource "ibm_kms_key_policies" "standard_key_policy" {
     enabled = var.dual_auth_delete_enabled
   }
 }
+
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  kmip_root_key_validation = (length(var.kmip) > 0 && var.standard_key) ? tobool("When providing a KMIP Adapter/Certificate for a key, the key being created must be a root key.") : true
```

Comment:
perhaps be more explicit in the error message, and include the exact variable and the value that needs to be set to provision a root key as opposed to a standard key